src/main/java/org/springframework/hateoas/core/JsonPathLinkDiscoverer.java

```diff
@@ -178,6 +177,16 @@ private JsonPath getExpression(String rel) {
 	 */
 	@Override
 	public boolean supports(MediaType delimiter) {
-		return this.mediaType == null ? true : this.mediaType.isCompatibleWith(delimiter);
+		boolean supports = false;
+		if (this.mediaTypes == null) {
+			return true;
+		} else {
+			for (MediaType mediaType : this.mediaTypes) {
```

Comment:
Wasn't sure what to do here either with the `List` of `MediaType` and `isCompatibleWith`